code/modules/mob/transform_procs.dm

```diff
@@ -183,85 +184,52 @@
 
 
 //human -> robot
-/mob/living/carbon/human/proc/Robotize(var/delete_items = 0, var/skipnaming=0)
-	if (monkeyizing)
-		return
-	for(var/obj/item/W in src)
-		if(delete_items)
-			qdel(W)
-		else
-			drop_from_inventory(W)
-	dropBorers()
-	regenerate_icons()
-	monkeyizing = 1
-	canmove = 0
-	icon = null
-	invisibility = 101
-	delayNextAttack(50)
-	for(var/t in organs)
-		qdel(t)
+/mob/proc/Robotize(var/delete_items = FALSE, var/skipnaming=FALSE)
+	Premorph(delete_items)
 
 	var/mob/living/silicon/robot/O = new /mob/living/silicon/robot(get_turf(src))
 	. = O
-	// cyborgs produced by Robotize get an automatic power cell
-	O.cell = new(O)
-	O.cell.maxcharge = 7500
-	O.cell.charge = 7500
-
-	O.setGender(gender)
-	O.invisibility = 0
 
 	if(mind)		//TODO
 		mind.transfer_to(O)
 		if(O.mind.assigned_role == "Cyborg")
 			O.mind.original = O
-		else if(mind&&mind.special_role)
+		else if(mind && mind.special_role)
 			O.mind.store_memory("In case you look at this after being borged, the objectives are only here until I find a way to make them not show up for you, as I can't simply delete them without screwing up round-end reporting. --NeoFite")
 	else
 		O.key = key
 
 	O.forceMove(loc)
-	O.job = "Cyborg"
-
 	O.mmi = new /obj/item/device/mmi(O)
 	O.mmi.transfer_identity(src)//Does not transfer key/client.
 
+	if(jobban_isbanned(O, "Cyborg")) //You somehow managed to get borged, congrats.
+		to_chat(src, "<span class='warning' style=\"font-family:Courier\">WARNING: Illegal operation detected.</span>")
+		to_chat(src, "<span class='danger'>Self-destruct mechanism engaged.</span>")
+		O.self_destruct()
+		message_admins("[key_name(O)] was forcefully transformed into a [job] and had its self-destruct mechanism engaged.")
+		log_game("[key_name(O)] was forcefully transformed into a [job] and had its self-destruct mechanism engaged.")
```

Comment:
Maybe you could add a comment that `O` was jobbanned for clarity of the message.